app/models/miq_worker.rb

```diff
@@ -267,6 +267,8 @@ def self.start_worker(*params)
     w
   end
 
+  cache_with_timeout(:my_worker) { server_scope.find_by(:pid => Process.pid) }
```

Comment:
I think this is ok, but it's kinda weird if subclasses can return rows of not of it's type.